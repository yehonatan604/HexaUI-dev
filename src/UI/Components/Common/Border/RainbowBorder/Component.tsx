import { paddingHelper } from "../../../../../Core/Helpers/Padding.helper";
import useTheme from "../../../../../Core/Hooks/useTheme";
import { TRainbowBorder } from "../../../../../Data/Types/ComponentTypes/Common/Rainbow-border/TRainbowBorder";

const RainbowBorder = (props: TRainbowBorder) => {
  const { children, options, ...componentProps } = props;

  //Hooks
  const { gradients } = useTheme();

  // Options
  const borderThickness = paddingHelper(options.thickness ?? 1.5);
  const gradient = options.gradient ?? gradients.primary;

  // JSX
  return (
    <div
      className={`${borderThickness} rounded bg-gradient-to-br ${gradient}`}
      {...componentProps}
    >
      {children}
    </div>
  );
};

export default RainbowBorder;
