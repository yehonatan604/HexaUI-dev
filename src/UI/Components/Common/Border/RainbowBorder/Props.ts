import { TComponent } from "../../../../../Data/Types/ComponentTypes/TComponent";

export type RainbowBorderProps = TComponent<"div"> & {
    variant?: string;
    borderWidth?: number;
};