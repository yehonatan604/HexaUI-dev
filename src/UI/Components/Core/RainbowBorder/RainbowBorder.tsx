import useTheme from "../../../../Core/Context/ThemeContext/hooks/useTheme";
import { TRainbowBorder } from "./types/TRainbowBorder";

const RainbowBorder = (props: TRainbowBorder) => {
  const { children, options, className, innerClassName, ...componentProps } = props;

  //Hooks
  const { gradients, colors } = useTheme();

  // Options
  const borderThickness = options?.thickness || "[1.5px]";
  const gradient = options?.gradient || gradients.primary;
  const rounded = `${options?.rounded ? "rounded-" + options?.rounded : "rounded"}`;

  // Inner Options
  const innerBgColor = options?.innerBgVariant || colors.black;
  const innerTextColor = options?.innerTextVariant || colors.white;
  const innerRounded = `${
    options?.innerRounded
      ? "rounded-" + options?.innerRounded
      : options?.rounded
      ? "rounded-" + options?.rounded
      : "rounded"
  }`;

  // JSX
  return (
    <div
      className={`p-${borderThickness} ${rounded} bg-gradient-to-br ${gradient} ${className}`}
      {...componentProps}
    >
      <div
        className={`text-${innerTextColor} bg-${innerBgColor} ${innerRounded} ${innerClassName}`}
      >
        {children}
      </div>
    </div>
  );
};

export default RainbowBorder;
