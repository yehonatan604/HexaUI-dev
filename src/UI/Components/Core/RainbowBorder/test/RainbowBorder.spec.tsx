import { render } from "@testing-library/react";
import RainbowBorder from "../RainbowBorder";
import { test, expect, describe } from "vitest";

describe("RainbowBorder Component", () => {
  test("matches snapshot", () => {
    const { container } = render(<RainbowBorder>Content</RainbowBorder>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders the RainbowBorder component correctly", () => {
    const { container } = render(<RainbowBorder>Content</RainbowBorder>);
    expect(container.firstChild).toBeInTheDocument();
  });

  test("applies custom thickness and gradient classes", () => {
    const { container } = render(
      <RainbowBorder
        options={{ thickness: "3px", gradient: "from-blue-500 to-green-500" }}
      >
        Content
      </RainbowBorder>
    );
    expect(container.firstChild).toHaveClass("p-3px");
    expect(container.firstChild).toHaveClass(
      "bg-gradient-to-br from-blue-500 to-green-500"
    );
  });

  test("applies default thickness and gradient when options are not provided", () => {
    const { container } = render(<RainbowBorder>Content</RainbowBorder>);
    expect(container.firstChild).toHaveClass("p-[1.5px]");
    expect(container.firstChild).toHaveClass("bg-gradient-to-br");
  });

  test("renders inner element with correct classes", () => {
    const { container } = render(
      <RainbowBorder options={{ innerBgVariant: "red-500", innerRounded: "lg" }}>
        Inner Content
      </RainbowBorder>
    );
    const innerElement = container.firstChild?.firstChild;
    expect(innerElement).toHaveClass("bg-red-500");
    expect(innerElement).toHaveClass("rounded-lg");
  });

  test("renders children inside the RainbowBorder", () => {
    const { getByText } = render(
      <RainbowBorder>
        <p>Test Content</p>
      </RainbowBorder>
    );
    expect(getByText("Test Content")).toBeInTheDocument();
  });
});
