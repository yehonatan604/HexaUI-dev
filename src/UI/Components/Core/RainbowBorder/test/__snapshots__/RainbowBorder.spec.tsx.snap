// Vitest Snapshot v1

exports[`RainbowBorder Component > matches snapshot 1`] = `
<div
  class="p-[1.5px] rounded bg-gradient-to-br from-blue-400 via-blue-600 via-blue-800 via-green-500 via-green-800 to-cyan-400 undefined"
>
  <div
    class="text-white bg-black rounded undefined"
  >
    Content
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  class="p-[1.5px] rounded bg-gradient-to-br from-blue-400 via-blue-600 via-blue-800 via-green-500 via-green-800 to-cyan-400 undefined"
>
  <div
    class="text-white bg-black rounded undefined"
  >
    Content
  </div>
</div>
`;
